File: packages/engine/src/cpus/6502/instructions/iny.js
import { CycleType } from '..'

/**
 * INcrement Y
 *
 * Y,Z,N = Y+1
 *
 * Affects Flags: N Z
 *
 * @example
 * MODE          SYNTAX        HEX LEN TIM
 * Implied       INY           $C8  1   2
 */
export function* iny() {
  if (++this.y > 255) this.y = 0
  this.zero = this.y === 0
  this.negative = (this.y & 0x80) > 0
  yield CycleType.GET
}

File: packages/engine/src/cores/famibob/cpu/instructions/iny.zig
const std = @import("std");
const CPU = @import("../cpu.zig").CPU;
const Addressing = @import("../opcode.zig").Addressing;

// INcrement Y
//
// Y,Z,N = Y+1
//
// Affects Flags: N Z
//
// MODE          SYNTAX        HEX LEN TIM
// Implied       INY           $C8  1   2

pub fn iny(self: *CPU, addressing: Addressing) void {
    _ = addressing;
    self.y +%= 1;
    self.z = self.y == 0;
    self.n = (self.y & 0x80) > 0;
}
